### PR TITLE
[Tooling] Use pagination for code freeze

### PR DESCRIPTION
**Pull request by pierotibou**
## Summary of changes

Start and end freeze actions now handle PRs by group of 25.

## Reason for change
It was often failing and needed retries

## Implementation details
Added a new shared action used by both start and end code freeze workflows.
The action handles 25 PRs and is called page by page

## Test coverage
Tested both starting and ending the freeze from the branch. It works

### CODE REVIEW

**Review by bouwkast — APPROVED** [review]
Looks good to me, just a question on one of the secret names

File: .github/workflows/code_freeze_end.yml [hunk 1]
```diff
@@ -9,12 +9,15 @@ jobs:
...
       github.event_name == 'workflow_dispatch' 
-      || (github.event.milestone.title == 'Code Freeze' && github.event.milestone.state == 'closed') 
+      || (github.event.milestone.title == 'Code Freeze' && github.event.milestone.state == 'closed')
     runs-on: ubuntu-latest
     env:
       GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
 
     steps:
+    - name: Clone dd-trace-dotnet repository
+      uses: actions/checkout@v3
```
Comment (andrewlock) [inline .github/workflows/code_freeze_end.yml:19]: TBH, if we're going to clone the repo anyway, I would avoid the bash and just do it in Nuke

File: .github/workflows/code_freeze_start.yml [hunk 2]
```diff
@@ -9,65 +9,43 @@ jobs:
...
-    - name: Trigger AAS deploy
-      run: |
-        curl -X POST \
-        -H "Accept: application/vnd.github.v3+json" \
-        -H "Authorization: Bearer ${{ secrets.GH_TOKEN_PIERO }}" \
-        https://api.github.com/repos/DataDog/datadog-aas-extension/dispatches \
-        -d '{"event_type": "dd-trace-code-freeze", "client_payload": {"sha":"'"$GITHUB_SHA"'" } }'
\ No newline at end of file
+          -H "Accept: application/vnd.github.v3+json" \
+          -H "Authorization: Bearer ${{ secrets.GH_TOKEN_PIERO }}" \
```
[the remark this answers is not in the document]
Comment (andrewlock) [reply .github/workflows/code_freeze_start.yml:49]: I don't _think_ so, as that's what it was previously 😅

File: .github/workflows/code_freeze_end.yml [hunk 3]
```diff
@@ -27,42 +30,30 @@ jobs:
...
+        page_number: 3
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        end_freeze: "true"
 
-        
-        
+    - uses: ./.github/actions/code-freeze
+      name: 'Unfreeze 25 PRs'
+      with:
+        page_number: 4
```
[the remark this answers is not in the document]
Comment (pierotibou) [reply .github/workflows/code_freeze_end.yml:57]: We add a new call with page 5 :). We had the same problem with the previous one so we can improve later :p

**Pull request merged**